src/components/EggheadPlayer/use-egghead-player.ts

```diff
@@ -0,0 +1,213 @@
+import cookies from '@utils/cookies'
+import {track} from '@utils/analytics'
+import getAccessTokenFromCookie from '@utils/getAccessTokenFromCookie'
+import axios from 'axios'
+import {isEmpty, pickBy, identity, get} from 'lodash'
+
+const getOptions = () =>
+  getAccessTokenFromCookie()
+    ? {
+        headers: {
+          Authorization: `Bearer ${getAccessTokenFromCookie()}`,
+        },
+      }
+    : {}
+
+const toLessonViewParams = ({collection, lesson}) => {
+  if (isEmpty(lesson)) {
+    console.error('Cannot track lesson view for undefined lesson.')
+    return
+  }
+
+  return {
+    lesson_view: pickBy(
+      {
+        collection_type: get(collection, 'type'),
+        collection_id: get(collection, 'id'),
+        lesson_id: lesson.id,
+      },
+      identity,
+    ),
+  }
+}
+
+let lessonProgress
+
+const createLessonView = (lesson, collection?) => {
+  return axios
+    .post(
+      lesson.lesson_view_url,
+      toLessonViewParams({lesson, collection}),
+      getOptions(),
+    )
+    .then(({data}) => {
+      const lessonView = data
+      return lessonView
+    })
+}
+
+const incrementViewSegments = (lessonView) => {
+  return axios
+    .put(lessonView.increment_url, {}, getOptions())
+    .then(({data}) => data)
+}
+
+const storeProgress = (playedSeconds, lesson, list?) => {
+  return new Promise((resolve) => {
+    const isSegment = Math.round(playedSeconds) % 30 === 0
+    if (isSegment) {
+      createLessonView(lesson, list).then(incrementViewSegments).then(resolve)
+    }
+  })
+}
+
+const onProgress = (lesson) => (progress) => {
```

Comment:
logically this ported over mostly intact. It's missing some details around playlists (collections) that is used primarily to decide what to show in the "next up list" but also as metadata to analytics.